packages/runner-ct/src/app/RunnerCt.tsx

```diff
@@ -54,235 +81,129 @@ const App: React.FC<AppProps> = observer(
-        interaction: {
-          type: 'anchor',
-          href: 'https://on.cypress.io/component-testing',
-          onClick: ({ event, index }) => {
-            if (!event.currentTarget || !event.currentTarget.href) {
-              return
-            }
-
-            event.preventDefault()
-            props.eventManager.reporterBus.emit('external:open', event.currentTarget.href)
-          },
-        },
-      },
-    ]
+      props.eventManager.saveState({ ctIsSpecsListOpen: newVal })
 
-    function toggleSpecsList () {
-      setActiveIndex((val) => val === 0 ? undefined : 0)
-      const newVal = !props.state.isSpecsListOpen
+      return newVal
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.eventManager])
 
-      state.setIsSpecsListOpen(newVal)
-      props.eventManager.saveState({ ctIsSpecsListOpen: newVal })
-    }
+    const navItems = React.useMemo(() =>
+      buildNavItems(props.eventManager, toggleIsSpecsListOpen)
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    , [props.eventManager, toggleIsSpecsListOpen])
 
-    function focusSpecsList () {
-      setActiveIndex(0)
-      state.setIsSpecsListOpen(true)
+    const focusSpecsList = React.useCallback(() => {
+      toggleIsSpecsListOpen(true)
 
       // a little trick to focus field on the next tick of event loop
       // to prevent the handled keydown/keyup event to fill input with "/"
       setTimeout(() => {
         searchRef.current?.focus()
       }, 0)
-    }
+    }, [toggleIsSpecsListOpen])
 
-    useGlobalHotKey('ctrl+b,command+b', toggleSpecsList)
+    useGlobalHotKey('ctrl+b,command+b', toggleIsSpecsListOpen)
     useGlobalHotKey('/', focusSpecsList)
 
-    function onReporterSplitPaneChange (newWidth: number) {
-      state.updateReporterWidth(newWidth)
-    }
-
-    function onPluginsSplitPaneChange (newHeight: number) {
-      state.updatePluginsHeight(newHeight)
-    }
-
-    function onSpecListPaneChange (newWidth: number) {
-      state.updateSpecListWidth(newWidth)
-    }
+    useScreenshotHandler({
+      state,
+      eventManager,
+      splitPaneRef,
+    })
 
-    function persistWidth (prop: 'ctReporterWidth' | 'ctSpecListWidth') {
+    // Inner function should probably be memoed, but I will avoid it until we see data requiring it
+    const persistWidth = (prop: 'ctReporterWidth' | 'ctSpecListWidth') => {
       return (newWidth: number) => {
         props.eventManager.saveState({ [prop]: newWidth })
       }
     }
 
-    function hideIfScreenshotting (callback: () => number) {
-      if (state.screenshotting) {
-        return 0
-      }
-
-      return callback()
-    }
+    const filteredSpecs = props.state.specs.filter((spec) => spec.relative.toLowerCase().includes(search.toLowerCase()))
 
-    function hideReporterIfNecessary (callback: () => number) {
-      if (state.screenshotting || !state.spec) {
-        return 0
+    React.useLayoutEffect(() => {
+      if (config.isTextTerminal) {
```

Comment:
This should not be a part of component at all, ideally should be set to false when `new State` called

Reply:
Hm... You're right. Fixed.

At some point we should refactor so `config` doesn't have to be passed everywhere. I tried and found that Cypress driver code currently mutates it, so it can't be a MobX observable.